frontend/src/scenes/insights/insightLogic.ts

```diff
@@ -356,6 +356,12 @@ export const insightLogic = kea<insightLogicType>({
                 ...insight,
             }),
             setInsightMetadata: (state, { metadata }) => ({ ...state, ...metadata }),
+            [dashboardsModel.actionTypes.updateDashboardItem]: (state, { item }) => {
+                if (item.short_id === state.short_id) {
+                    return { ...state, dashboards: item.dashboards }
+                }
+                return state
+            },
```

Comment:
I wonder if we shouldn't just update everything? Any chance we need to? What could go wrong if we do?

Reply:
My understanding was that this is being called for every `updateDashboardItem` not just those for this insight/dashboard combo. I'll confirm that